src/assisted_test_infra/test_infra/helper_classes/config/base_config.py

```diff
@@ -1,35 +1,12 @@
-from abc import ABC, abstractmethod
+from abc import ABC
 from dataclasses import asdict, dataclass
-from typing import Any, Dict, Tuple
+from typing import Any
 
-from service_client import log
-
-
-class Triggerable(ABC):
-    def trigger(self, triggers: Dict[Tuple[Tuple[str, Any]], Dict[str, Any]]):
-        for conditions, values in triggers.items():
-            assert isinstance(conditions, tuple) and all(
-                isinstance(condition, tuple) for condition in conditions
-            ), f"Key {conditions} must be tuple of tuples"
-
-            if all(self._is_set(param, expected_value) for param, expected_value in conditions):
-                self._handle_trigger(conditions, values)
-
-    def _is_set(self, var, expected_value):
-        return getattr(self, var, None) == expected_value
-
-    def _handle_trigger(self, conditions: Tuple[Tuple[str, Any]], values: Dict[str, Any]) -> None:
-        for k, v in values.items():
-            self._set(k, v)
-        log.info(f"{conditions} is triggered. Updating global variables: {values}")
-
-    @abstractmethod
-    def _set(self, key: str, value: Any):
-        pass
+from triggers.env_trigger import Triggerable
 
 
 @dataclass
-class _BaseConfig(Triggerable, ABC):
+class BaseConfig(Triggerable, ABC):
```

Comment:
`_get_data_pool` is only exists on the configurations implementations and not in `BaseConfig`

Reply:
so can you add it explicitly to this class?
```
@abstractmethod
def _get_data_pool(self, ...):
    pass
```

Reply:
No, because that `_get_data_pool` is more of implementation of this class and it's accessing the environment variables that we removed the access to it few months ago.